### PR TITLE
improvements

### PR DESCRIPTION
**Pull request by dorin-iancu**
- remove refund callback logic, it would fail on 99% of cases anyway
- use transfer&execute instead of async call
- other minor optimizations

### CODE REVIEW

File: .github/workflows/actions.yml [hunk 1]
```diff
@@ -21,8 +21,8 @@ jobs:
           source .github/workflows/env
           pip3 install erdpy
           mkdir $HOME/elrondsdk
-          erdpy config set dependencies.arwentools.tag v1.3.3
-          erdpy deps install arwentools
+          erdpy config set dependencies.arwentools.tag v1.4.27
+          erdpy deps install vmtools
```
Comment (camilbancioiu) [inline .github/workflows/actions.yml:25]:
```suggestion
          erdpy deps install all
```

**Review by camilbancioiu — CHANGES_REQUESTED** [review]

File: egld-esdt-swap/src/lib.rs [hunk 2]
```diff
@@ -24,79 +24,84 @@ pub trait EgldEsdtSwap {
...
 
     #[payable("*")]
     #[endpoint(unwrapEgld)]
     fn unwrap_egld(
         &self,
-        #[payment] payment: Self::BigUint,
-        #[payment_token] token_id: TokenIdentifier,
+        #[payment_token] payment_token: TokenIdentifier,
+        #[payment_amount] payment_amount: Self::BigUint,
         #[var_args] accept_funds_endpoint_name: OptionalArg<BoxedBytes>,
```
Comment (andrei-marinica) [inline egld-esdt-swap/src/lib.rs:69]: use ManagedBuffer instead of BoxedBytes

File: egld-esdt-swap/src/lib.rs [hunk 3]
```diff
@@ -24,79 +24,84 @@ pub trait EgldEsdtSwap {
     #[endpoint(wrapEgld)]
     fn wrap_egld(
         &self,
-        #[payment] payment: Self::BigUint,
+        #[payment_token] payment_token: TokenIdentifier,
+        #[payment_amount] payment_amount: Self::BigUint,
         #[var_args] accept_funds_endpoint_name: OptionalArg<BoxedBytes>,
```
Comment (andrei-marinica) [inline egld-esdt-swap/src/lib.rs:29]: use ManagedBuffer instead of BoxedBytes

**Review by andrei-marinica — APPROVED** [review]

File: egld-esdt-swap/src/lib.rs (hunk 2 above)
Comment (andrei-marinica) [reply egld-esdt-swap/src/lib.rs:69]: oh, now I see you are on 0.18. Nevermind.

**Review by sasurobert — COMMENTED** [review]
only what Andrei commented

**Pull request merged**